src/formatDate.js

```diff
@@ -50,7 +50,34 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateSplitted = date.split(fromFormat[3]);
+  const oldDate = {};
+  const resultDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    oldDate[fromFormat[i]] = dateSplitted[i];
+  }
+
+  for (const key of toFormat) {
+    if (oldDate[key]) {
+      resultDate.push(oldDate[key]);
+      continue;
+    }
+
+    if (oldDate['YY']) {
+      const normalizedYear
+        = (oldDate['YY'] >= 30 ? '19' : '20') + oldDate['YY'];
```

Comment:
It is a bad idea to mix ternary operator and other calculations in one line.
It works, but it makes code less readable.